src/kakarot/accounts/library.cairo

```diff
@@ -15,64 +18,311 @@ from starkware.starknet.common.syscalls import (
     storage_read,
     storage_write,
     StorageReadRequest,
+    CallContract,
+    get_tx_info,
+    get_contract_address,
+    get_caller_address,
+    replace_class,
 )
 from starkware.cairo.common.memset import memset
+from utils.eth_transaction import EthTransaction
 
 from kakarot.interfaces.interfaces import IERC20, IKakarot
+from kakarot.errors import Errors
 
+// @dev: should always be zero for EOAs
 @storage_var
-func bytecode_len_() -> (res: felt) {
+func Account_bytecode_len() -> (res: felt) {
 }
 
 @storage_var
-func storage_(key: Uint256) -> (value: Uint256) {
+func Account_storage(key: Uint256) -> (value: Uint256) {
 }
 
 @storage_var
-func is_initialized_() -> (res: felt) {
+func Account_is_initialized() -> (res: felt) {
 }
 
 @storage_var
-func evm_address() -> (evm_address: felt) {
+func Account_evm_address() -> (evm_address: felt) {
 }
 
 @storage_var
-func nonce() -> (nonce: felt) {
+func Account_nonce() -> (nonce: felt) {
+}
+
+@storage_var
+func Account_kakarot_address() -> (kakarot_address: felt) {
+}
+
+@storage_var
+func Account_implementation() -> (address: felt) {
+}
+
+@event
+func transaction_executed(response_len: felt, response: felt*, success: felt, gas_used: felt) {
 }
 
-// Define the number of bytes per felt
 const BYTES_PER_FELT = 31;
 
-// @title ContractAccount main library file.
-// @notice This file contains the EVM smart contract account representation logic.
-namespace ContractAccount {
+// @title Account main library file.
+// @notice This file contains the EVM account representation logic.
+// @dev: Both EOAs and Contract Accounts are represented by this contract.
+namespace GenericAccount {
     // @notice This function is used to initialize the smart contract account.
     func initialize{
         syscall_ptr: felt*,
         pedersen_ptr: HashBuiltin*,
         range_check_ptr,
         bitwise_ptr: BitwiseBuiltin*,
     }(kakarot_address: felt, _evm_address) {
-        let (is_initialized) = is_initialized_.read();
+        let (is_initialized) = Account_is_initialized.read();
         assert is_initialized = 0;
-        is_initialized_.write(1);
+        Account_is_initialized.write(1);
         Ownable.initializer(kakarot_address);
-        evm_address.write(_evm_address);
+        Account_evm_address.write(_evm_address);
+        Account_kakarot_address.write(kakarot_address);
+
         // Give infinite ETH transfer allowance to Kakarot
         let (native_token_address) = IKakarot.get_native_token(kakarot_address);
         let (infinite) = uint256_not(Uint256(0, 0));
```

Comment:
we can just hardcode `Uint256(0xffff..fff, 0xff.ff)`, maybe as a constant if you prefer like
```
const UINT256_MAX = ...
```
there may be one sowhere already